### PR TITLE
Implement decoder-driven sampling

### PR DESCRIPTION
**Pull request by xuxu-wei**
## Summary
- store training latent statistics and draw decoder latents for unconditional and conditional sampling
- translate decoder outputs into schema-aligned DataFrames via new sampling helpers
- expand tests covering schema-shaped samples and conditional label validation

## Testing
- pytest -q

------
https://chatgpt.com/codex/tasks/task_e_68cb8b95094c8320ba840aaf717abfaa

### CODE REVIEW

File: suave/sampling.py [hunk 1]
```diff
@@ -1,22 +1,91 @@
...
+        elif spec.type == "cat":
+            codes = sample_tensor.to(torch.int64).cpu().numpy()
+            categories = norm_stats.get(column, {}).get("categories")
+            data[column] = _categorical_from_codes(codes, categories)
+        elif spec.type == "ordinal":
+            thermometer = sample_tensor
+            levels = (thermometer > 0.5).sum(dim=-1)
+            codes = torch.clamp(levels - 1, min=0).to(torch.int64).cpu().numpy()
+            categories = norm_stats.get(column, {}).get("categories")
+            data[column] = _categorical_from_codes(codes, categories, ordered=True)
```
[the remark this answers is not in the document]
Comment (xuxu-wei) [reply suave/sampling.py:88]: @codex 我认为在这种情况下，扩展类别列表并抛出一个警告信息是合适的默认处理方式。
